### PR TITLE
Migrate to Next.Js

### PR DESCRIPTION
**Pull request by Comeza**

### CODE REVIEW

File: lib/database.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import * as admin from "firebase-admin";
+import {getFirestore} from "firebase-admin/firestore";
+import {Article} from "../models/article";
```
Comment (Comeza) [inline lib/database.ts:3]: Using absolute imports would make it easier to refractor code later.

File: lib/database.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const document = await db.collection("articles").add(article);
+    return document.id;
+}
+
+/**
+ * Fetches the article with the given id.
+ * @param id the id of the article
+ * @returns the article with the given id, or null if no article with the given id exists
+ */
+export async function getArticle(id: string): Promise<Article | null> {
```
Comment (Comeza) [inline lib/database.ts:29]: I think it would be prettier to fail the promise if there is no data.

File: lib/database.ts (hunk 1 above)
Comment (ytausch) [reply lib/database.ts]: Done.

File: lib/database.ts (hunk 2 above)
Comment (ytausch) [reply lib/database.ts:29]: Not sure about this one. I think failing the promise blows up the code if we still want proper error handling. I've created a [patch file](https://github.com/Comeza/lore/files/8465667/fail_the_promise_instead_of_returning_null_in_getArticle.id__string.patch.txt) of how I would implement this.

Returning null should be [perfectly fine](https://stackoverflow.com/a/1626620/13775379) IMO.

**Pull request merged**